blog/views.py

```diff
@@ -56,3 +55,70 @@ class EditPost(UpdateView, Protected):
     model = Post
     fields = ['title', 'text']
     template_name = 'blog/post_edit.html'
+
+
+class PostDraftList(ListView, Protected):
+    """Return draft list."""
+
+    queryset = Post.objects.filter(published_date__isnull=True).order_by('created_date')
+    context_object_name = 'posts'
+    template_name = 'blog/post_draft_list.html'
+
+
+class PublishPost(Protected, TemplateView):
+    """View for publishing post."""
+
+    def get(self, request, *args, **kwargs):
+        """Get info about POst pk that we use."""
+        post = get_object_or_404(Post, pk=kwargs['pk'])
+        post.publish()
+        return redirect('post_detail', pk=post.pk)
+
+
+class RemovePost(Protected, DeleteView):
+    """View for deleting posts."""
+
+    model = Post
+    success_url = reverse_lazy('post_list')
+    template_name = 'blog/post_edit.html'
+
+
+class AddCommentToPost(FormView):
+    """If someone wants to create new comment he/she get this view."""
+
+    form_class = CommentForm
+    template_name = 'blog/post_edit.html'
+
+    def post(self, request, *args, **kwargs):
+        """Add comment to DB."""
+        post = get_object_or_404(Post, pk=kwargs['pk'])
+        form = CommentForm(request.POST)
+        if form.is_valid():
+            comment = form.save(commit=False)
+            comment.post = post
+            comment.save()
+            return redirect('post_detail', pk=post.pk)
+
+
+class ApproveComment(Protected, TemplateView):
```

Comment:
TemplateView shouldn't be used for redirects, but for showing some representation.